pkg/syslog/rfc6587/writer.go

```diff
@@ -0,0 +1,18 @@
+package rfc6587
+
+import (
+	"fmt"
+	"io"
+)
+
+type Writer struct {
+	w io.Writer
+}
+
+func NewWriter(w io.Writer) *Writer {
+	return &Writer{w}
+}
+
+func (w *Writer) Write(p []byte) (int, error) {
+	return fmt.Fprintf(w.w, "%d %s", len(p), p)
```

Comment:
yeah, I started off with a `type WriterFunc func([]byte) (int, error)` a la `http.HandlerFunc`, but I later switched to `fmt.Fprintf` and that broke it. It's a smell but it shouldn't stick around long.